node/src/components/chain_synchronizer/operations.rs

```diff
@@ -740,156 +796,206 @@ async fn archival_sync(
                 continue;
             }
         };
-        sync_deploys_and_transfers_and_state(effect_builder, &most_recent_block, &node_config)
-            .await?;
+        sync_deploys_and_transfers_and_state(
+            *ctx.effect_builder,
+            &most_recent_block,
+            ctx.node_config,
+        )
+        .await?;
         if let Some(key_block_info) = KeyBlockInfo::maybe_from_block_header(
```

Comment:
(Then this could be `cts.maybe_key_block_info(most_recent_block.header())`.)

Reply:
See here: https://github.com/casper-network/casper-node/pull/2664#discussion_r807038024